docs/userguide/searchspace.md

```diff
@@ -1,46 +1,234 @@
-# Search spaces
-
-## General information
+# Search Spaces
```

Comment:
@RimRihana since you worked quite a lot with search spaces during your last PRs I think your feedback/review on this user guide would be very valuable. In particular - is there something that you feel should be added or is misleading?

Reply:
Sure! I will give you my review soon.